frontend/src/views/WheresMyGene/components/HeatMap/index.tsx

```diff
@@ -1,277 +1,95 @@
+import { State } from "../../common/store";
+import { CellTypeSummary, GeneExpressionSummary } from "../../common/types";
+import Chart from "./components/Chart";
+import XAxisChart from "./components/XAxisChart";
+import YAxisChart from "./components/YAxisChart";
+import { ChartWrapper, Container, Loader, YAxisWrapper } from "./style";
+import { getHeatmapHeight, X_AXIS_CHART_HEIGHT_PX } from "./utils";
 
 interface Props {
-  cellTypes: CellTypeSummary[];
-  data: CellTypeSummary[];
+  cellTypes: { [tissue: string]: CellTypeSummary[] };
   genes: State["selectedGenes"];
   tissuesWithDeletedCellTypes: string[];
-  allTissueCellTypes: { [region: string]: CellTypeSummary[] };
+  allTissueCellTypes: { [tissue: string]: CellTypeSummary[] };
+  selectedGeneData: GeneExpressionSummary[];
 }
 
-const ELEMENT_ID = "heat-map";
-
-let isChartInitialized = false;
-
-export default function HeatMap({
+export default memo(function HeatMap({
   cellTypes,
-  data,
   genes,
   tissuesWithDeletedCellTypes,
   allTissueCellTypes,
+  selectedGeneData,
 }: Props): JSX.Element {
-  const [chart, setChart] = useState<echarts.ECharts | null>(null);
-  const [xAxisChart, setXAxisChart] = useState<echarts.ECharts | null>(null);
-  const [yAxisChart, setYAxisChart] = useState<echarts.ECharts | null>(null);
-  const [isEchartGLAvailable, setIsEchartGLAvailable] = useState(false);
-  const [heatmapWidth, setHeatmapWidth] = useState(getHeatmapWidth(genes));
-  const [heatmapHeight, setHeatmapHeight] = useState(
-    getHeatmapHeight(cellTypes)
-  );
-
-  const [chartProps, setChartProps] = useState<ChartProps | null>(null);
-
-  const {
-    cellTypeIdsToDelete,
-    genesToDelete,
-    handleCellTypeClick,
-    handleGeneClick,
-  } = useDeleteGenesAndCellTypes();
-
-  const dispatch = useContext(DispatchContext);
-
-  // Loading state
-  const [isLoading, setIsLoading] = useState(false);
+  // Loading state per tissue
+  const [isLoading, setIsLoading] = useState(setInitialIsLoading(cellTypes));
 
   useEffect(() => {
-    setIsLoading(true);
+    setIsLoading((isLoading) => ({ ...isLoading, lung: true }));
   }, [genes, cellTypes]);
 
-  const debouncedDataToChartFormat = useMemo(() => {
-    return debounce(
-      (
-        data: CellTypeSummary[],
-        cellTypes: CellTypeSummary[],
-        genes: State["selectedGenes"]
-      ) => {
-        const result = {
-          cellTypeNames: getAllSerializedCellTypeMetadata(cellTypes),
-          chartData: dataToChartFormat(data, cellTypes, genes),
-          geneNames: getGeneNames(genes),
-        };
-
-        setChartProps(result);
-
-        setIsLoading(false);
+  const yAxisWrapperHeight = useMemo(() => {
+    const yAxisChartHeight = Object.values(cellTypes).reduce(
+      (height, cellTypeSummaries) => {
+        return height + getHeatmapHeight(cellTypeSummaries);
       },
-      DEBOUNCE_MS,
-      { leading: false }
+      0
     );
-  }, []);
-
-  useEffect(() => {
```

Comment:
this is moved into `Chart` component, so we don't need to re-render XAxisChart and YAxisChart unnecessarily